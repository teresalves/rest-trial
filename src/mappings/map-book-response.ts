import { Book, BookResponse } from "./schemas/response-schema";

export function mapBookResponse(books: Book[]): BookResponse {
    const jsonResponse: BookResponse = { books: {}};
    books.map((book) => jsonResponse.books[book.id] = book);
    return jsonResponse;
}