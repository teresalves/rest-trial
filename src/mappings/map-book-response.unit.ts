import { idText } from "typescript";
import { mapBookResponse } from "./map-book-response";

describe('mapBookResponse', () => {
    const rows = [
        {
            id: '1',
            title: 'title',
            year: 2000
        },
        {
            id: '2',
            title: 'title2',
            year: 2001
        }
    ]
    
    it('should map db rows to proper json object', () => {
        const result = mapBookResponse(rows);
        expect(result).toMatchSnapshot();
    })
})