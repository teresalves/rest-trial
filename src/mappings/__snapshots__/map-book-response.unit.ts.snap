// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapBookResponse test 1`] = `
Object {
  "books": Object {
    "1": Object {
      "id": "1",
      "title": "title",
      "year": 2000,
    },
    "2": Object {
      "id": "2",
      "title": "title2",
      "year": 2001,
    },
  },
}
`;
